pkg/destroy/aws/iamhelpers.go

```diff
@@ -46,13 +42,16 @@ func (search *iamRoleSearch) find(ctx context.Context) (arns []string, names []s
 				if err != nil {
 					var awsErr awserr.Error
 					if errors.As(err, &awsErr) {
-						switch awsErr.Code() {
-						case ErrCodeAccessDeniedException, iam.ErrCodeNoSuchEntityException:
-							// Installer does not have access to this IAM role or the
-							// the role does not exist.
+						switch {
+						case awsErr.Code() == iam.ErrCodeNoSuchEntityException:
+							// The role does not exist.
 							// Ignore this IAM Role and donot report this error via
 							// lastError
 							search.unmatched[*role.Arn] = exists
+						case strings.Contains(err.Error(), "AccessDenied"):
+							// Installer does not have access to this IAM role
+							// Ignore this IAM Role and donot report this error via
+							// lastError
```

Comment:
Shouldn't you 
```suggestion
							// lastError
							search.unmatched[*role.Arn] = exists
```
to ignore the role as the comment says?

Reply:
@sadasu If I'm reading it correctly, you can avoid this API call if you ever see this role again: https://github.com/openshift/installer/pull/7180/files#diff-2cf3547918c79b5cad1244c173df556f9664596bb88d8bc2a6c888e681d8a038R36-R38